Chapter 1/4_Palindrome Permutation/PalindromePermutation.py

```diff
@@ -4,31 +4,25 @@
 
 def pal_perm(phrase):
     '''function checks if a string is a permutation of a palindrome or not'''
-    table = [0 for _ in range(ord('z') - ord('a') + 1)]
-    countodd = 0
-    for c in phrase:
-        x = char_number(c)
-        if x != -1:
-            table[x] += 1
-            if table[x] % 2:
-                countodd += 1
-            else:
-                countodd -= 1
+    phrase = phrase.replace(" ", "").lower()
+    dic = {}
+    for i in phrase:
+        if i in dic.keys():
+            dic[i] += 1
+        else:
```

Comment:
if you're going for a pythonic solution then you might consider using defaultdict or Counter